main/main.c

```diff
@@ -186,7 +186,7 @@ static PHP_INI_MH(OnChangeMemoryLimit)
 	if (new_value) {
 		PG(memory_limit) = zend_atol(ZSTR_VAL(new_value), ZSTR_LEN(new_value));
 	} else {
-		PG(memory_limit) = 1<<30;		/* effectively, no limit */
+		PG(memory_limit) = 1UL<<30;		/* effectively, no limit */
```

Comment:
It looks like `memory_limit` is a `zend_long` (https://github.com/php/php-src/blob/384f0f8a4c91c3fa1c2de4786f971197a0d52ee7/main/php_globals.h#L68), so using `Z_L(1)<<30` might be preferable.

Reply:
Well, you know what they say about doing it yourself... feel free to! I'm outie.